crates/bevy_sprite/src/sprite.rs

```diff
@@ -185,6 +188,51 @@ impl SpriteImageMode {
             SpriteImageMode::Sliced(..) | SpriteImageMode::Tiled { .. }
         )
     }
+
+    /// Returns [`ScalingMode`] if scale is presented or [`Option::None`] otherwise.
+    #[inline]
+    #[must_use]
+    pub const fn scale(&self) -> Option<ScalingMode> {
+        if let SpriteImageMode::Scale(scale) = self {
+            Some(*scale)
+        } else {
+            None
+        }
+    }
+}
+
+/// Represents various modes for proportional scaling of a texture.
+///
+/// Can be used in [`SpriteImageMode::Scale`].
+#[derive(Debug, Clone, Copy, PartialEq, Default, Reflect)]
+#[reflect(Debug)]
+pub enum ScalingMode {
```

Comment:
I think maybe this type might be better split up into different two enums, one `Fill/Fit` and the other `Start/Center/End`.

Reply:
> I think maybe this type might be better split up into different two enums, one `Fill/Fit` and the other `Start/Center/End`.

I don't think that will be convenient for the end user. There are only six variants and no more are planned to add I guess. That led to the single field in Sprite that corresponds to scaling, adding one more enum adds more complexity.

Then `SpriteImageMode::Scale` needs to be a struct enum variant with two fields, one for `Fill/Fit` and the second one for direction `Start/Center/End`. In my humble opinion, it will not bring convenience for the current `API`.